src/timesteppers.jl

```diff
@@ -49,4 +49,90 @@ function update_time(s::FerriteSolver{<:Any,<:FixedTimeStepper}, t, step, conver
         throw(ConvergenceError(msg))
     end
     return s.timestepper.t[step+1], step+1
-end
\ No newline at end of file
+end
+
+
+"""
+    AdaptiveTimeStepper(
+        Δt_init::T, t_end::T; 
+        t_start=zero(T), Δt_min=Δt_init, Δt_max=typemax(T), 
+        change_factor=T(1.5), num_converged_to_increase::Int=1) where T
+
+An adaptive time stepper, starting with a step `Δt_init` and with a 
+maximum time of `t_end`. The `change_factor` describes how much the 
+time step is changed in the case of (a) no convergence: 
+`Δt/=change_factor` and (b) converged the last `num_converged_to_increase`
+time steps: `Δt*=change_factor`.
+"""
+struct AdaptiveTimeStepper{T}
+    t_start::T
+    t_end::T
+    Δt_init::T
+    Δt_min::T
+    Δt_max::T
+    change_factor::T 
+    num_converged_to_increase::Int
+    num_converged::ScalarWrapper{Int}
```

Comment:
What's the benefit of `ScalarWrapper`? I simply don't know, but I do know that it's also present in Ferrite.jl

Reply:
I used it to avoid using the vector with one field. But we could just as well do that. 
Or just have the struct mutable, probably the cleanest solution. 
In 1.8+ (or 1.7 also maybe) we could add `const` annotation!

Reply:
ah I think I got it now, so you wanted to avoid having the `AdaptiveTimeStepper` being mutable and therefore introduced a scalar wrapper which is mutable, but doesnt need to be a vector - since it's just a scalar. That's the idea, isn't it? The const annotation is in case of setting `AdaptiveTimeStepper` mutable and annotate everything with const? 

Just wanted to understand the reasoning. The ScalarWrapper seems most reasonable for me, especially since its backwards compatible

Reply:
Yes, exactly! (In my own code, I called it MutableScalar instead, but I figured since basically all users are Ferrite users, using the same name makes sense)
OK, let's keep it then!